lite/valor_lite/detection/manager.py

```diff
@@ -817,45 +908,129 @@ def add_data(
             # cache labels and annotations
             keyed_groundtruths = defaultdict(list)
             keyed_predictions = defaultdict(list)
+
             for gidx, gann in enumerate(detection.groundtruths):
-                self._evaluator.groundtruth_examples[uid_index][
-                    gidx
-                ] = np.array(gann.extrema)
+                if not isinstance(gann, annotation_type):
+                    raise ValueError(
+                        f"Expected {annotation_type}, but annotation is of type {type(gann)}."
+                    )
+
+                if annotation_type == BoundingBox:
+                    self._evaluator.groundtruth_examples[uid_index][
+                        gidx
+                    ] = _get_annotation_representation(obj=gann)
+
                 for glabel in gann.labels:
                     label_idx, label_key_idx = self._add_label(glabel)
                     self.groundtruth_count[label_idx][uid_index] += 1
+                    representation = _get_annotation_representation(obj=gann)
                     keyed_groundtruths[label_key_idx].append(
                         (
                             gidx,
                             label_idx,
-                            gann.extrema,
+                            representation,
                         )
                     )
+
             for pidx, pann in enumerate(detection.predictions):
-                self._evaluator.prediction_examples[uid_index][
-                    pidx
-                ] = np.array(pann.extrema)
+                if not isinstance(pann, annotation_type):
+                    raise ValueError(
+                        f"Expected {annotation_type}, but annotation is of type {type(pann)}."
+                    )
+
+                if annotation_type == BoundingBox:
+                    self._evaluator.prediction_examples[uid_index][
+                        pidx
+                    ] = _get_annotation_representation(obj=pann)
                 for plabel, pscore in zip(pann.labels, pann.scores):
                     label_idx, label_key_idx = self._add_label(plabel)
                     self.prediction_count[label_idx][uid_index] += 1
+                    representation = _get_annotation_representation(
```

Comment:
Same as above, can you move this out of the label loop since it only needs to be run once per annotation?